include/openssl/crypto.h

```diff
@@ -84,6 +84,28 @@ OPENSSL_EXPORT int CRYPTO_has_broken_NEON(void);
 OPENSSL_EXPORT int CRYPTO_needs_hwcap2_workaround(void);
 #endif  // OPENSSL_ARM && OPENSSL_LINUX && !OPENSSL_STATIC_ARMCAP
 
+// Data-Independent Timing (DIT) on AArch64
+
+#if defined(OPENSSL_AARCH64) && !defined(OPENSSL_WINDOWS) && defined(MAKE_DIT_AVAILABLE)
+// (TODO): See if we can detect the DIT capability in Windows environment
+uint64_t armv8_get_dit(void);
+uint64_t armv8_enable_dit(void);
+void armv8_restore_dit(volatile uint64_t *original_dit);
+
+#define ENABLE_DIT_ALL                      \
+  volatile uint64_t _dit_orig                \
+          OPENSSL_UNUSED = armv8_enable_dit();
+
+
+#define ENABLE_DIT_AUTO_DISABLE                      \
+  volatile uint64_t _dit_restore_orig                \
+         __attribute__((cleanup(armv8_restore_dit))) \
+          OPENSSL_UNUSED = armv8_enable_dit();
```

Comment:
can you explain a bit what's this construction doing? I'm a bit lost tbh.

Reply:
From our offline conversation:
`_dit_restore_orig` on l.101 which is the current DIT value on scope entry is output by `armv8_enable_dit()`. On scope exit, it is passed to `armv8_restore_dit()`. To illustrate:
```
func() {
  // ENABLE_DIT_AUTO_DISABLE does the following:
  // This first part is done in armv8_enable_dit()
  old_dit_state <- read old dit state;
  if (old_dit_state != 1) {
   SET_DIT
  }
  ...
  // scope exit, calls armv8_restore_dit()
  if (old_dit_state != 1) UNSET_DIT;
}
```